src/backends/policy_enforcement/souffle/fact.h

```diff
@@ -0,0 +1,47 @@
+//-----------------------------------------------------------------------------
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//----------------------------------------------------------------------------
+
+#ifndef SRC_BACKENDS_POLICY_ENFORCEMENT_SOUFFLE_FACT_H_
+#define SRC_BACKENDS_POLICY_ENFORCEMENT_SOUFFLE_FACT_H_
+
+#include <utility>
+
+#include "src/backends/policy_enforcement/souffle/value.h"
+
+namespace raksha::backends::policy_enforcement::souffle {
+
+template <const char *kRelationName, class... RelationParameterTypes>
+class Fact {
```

Comment:
We should reconcile with `src/ir/datalog`. I don't want us to get too fragmented.

Reply:
Done.